index.js

```diff
@@ -56,7 +56,8 @@ module.exports = function(options) {
     let origin;
     if (typeof options.origin === 'function') {
       origin = options.origin(ctx);
-      if (origin instanceof Promise) origin = await origin;
+      const isAsync = origin.constructor.name === 'AsyncFunction';
```

Comment:
I'm not sure why we need this detect? When we call an async function, it will return a promise instance.

Reply:
@dead-horse , not always, the async function return a promise  instance.


```js
const p = new Promise(()=>{});
p instanceof Promise // true
const  a = async function () {}
a instanceof Promise // false
a.constructor.name === "AsyncFunction" // true 
```

chrome console
![image](https://user-images.githubusercontent.com/43971542/82147695-fe080c00-984f-11ea-9c6b-ef0b6530a815.png)

Reply:
```js
const  a = async function () {}
const result = a()
result instanceof Promise // true
```